packages/plugins/plugin-appsflyer/src/AppsflyerPlugin.tsx

```diff
@@ -3,27 +3,77 @@ import {
   IdentifyEventType,
   PluginType,
   TrackEventType,
+  UpdateType,
+  SegmentAPISettings,
 } from '@segment/analytics-react-native';
-import appsFlyer, { InitSDKOptions } from 'react-native-appsflyer';
+import type { SegmentAppsflyerSettings } from './types';
+import appsFlyer from 'react-native-appsflyer';
 import identify from './methods/identify';
 import track from './methods/track';
 
 export class AppsflyerPlugin extends DestinationPlugin {
   type = PluginType.destination;
-  key = 'Appsflyer';
+  key = 'AppsFlyer';
 
-  constructor(opts: InitSDKOptions) {
-    super();
+  private settings: SegmentAppsflyerSettings | null = null;
+  private hasRegisteredInstallCallback: Boolean = false;
+  private hasRegisteredDeepLinkCallback: Boolean = false;
```

Comment:
Should use lowercase `boolean` as the types with Uppercase are objects, not primitives